src/server/getters/get-reporting-fees.ts

```diff
@@ -62,6 +98,128 @@ interface RepStakeResults {
   lostRep: BigNumber;
 }
 
+function getUniverseAndParentUniverse(db: Knex, universe: Address|null, feeWindow: Address|null, callback: (err: Error|null, result?: UniverseAndParentUniverse) => void) {
+  const query = db("fee_windows")
+  .first("universes.universe", "universes.parentUniverse")
+  .join("universes", "fee_windows.universe", "universes.universe")
+  .groupBy("fee_windows.universe");
+  if (universe != null) {
+    query.where("fee_windows.universe", universe);
+  } else if (feeWindow != null) {
+    query.where("fee_windows.feeWindow", feeWindow);
+  }
+  query.asCallback((err, universeAndParentUniverse: UniverseAndParentUniverse) => {
+    if (err) return callback(err);
+    return callback(null, universeAndParentUniverse);
+  });
+}
+
+function formatMarketInfo(augur: Augur, marketParticipants: any, parentUniverse: Address) {
+  let forkedMarket: ForkedMarket|null = null;
+  const keyedNonforkedMarkets = {} as any;
+  let i: number;
+  for (i = 0; i < marketParticipants.initialReporters.length; i++) {
+    if (marketParticipants.initialReporters[i].forking && marketParticipants.initialReporters[i].universe === parentUniverse) {
+      forkedMarket = {
+        address: marketParticipants.initialReporters[i].marketId,
+        universeAddress: marketParticipants.initialReporters[i].universe,
+        isFinalized: marketParticipants.initialReporters[i].reportingState === augur.constants.REPORTING_STATE.FINALIZED,
+        crowdsourcers: [],
+        initialReporter: {
+          address: marketParticipants.initialReporters[i].initialReporter,
+          isForked: marketParticipants.initialReporters[i].disavowed ? true : false,
+        },
+      };
+    } else if (!marketParticipants.initialReporters[i].forking && marketParticipants.initialReporters[i].universe !== parentUniverse) {
+      keyedNonforkedMarkets[marketParticipants.initialReporters[i].marketId] = {
+        address: marketParticipants.initialReporters[i].marketId,
+        universeAddress: marketParticipants.initialReporters[i].universe,
+        crowdsourcersAreDisavowed: false,
+        isMigrated: !marketParticipants.initialReporters[i].needsMigration,
+        isFinalized: marketParticipants.initialReporters[i].reportingState === augur.constants.REPORTING_STATE.FINALIZED,
+        crowdsourcers: [],
+        initialReporterAddress: marketParticipants.initialReporters[i].initialReporter,
+      };
+    }
+  }
+  for (i = 0; i < marketParticipants.crowdsourcers.length; i++) {
+    if (marketParticipants.crowdsourcers[i].forking && marketParticipants.crowdsourcers[i].universe === parentUniverse) {
+      let forkedMarketIsEmpty: boolean = true;
+      if (forkedMarket) {
```

Comment:
Now that forked market is either null OR an object, we can just check it for `null` without iterating its keys

Reply:
Oops, good point. I forgot to take out the part where it's iterating over the keys. Will remove that.